src/filesystem-upload-options.js

```diff
@@ -11,14 +11,30 @@ governing permissions and limitations under the License.
 */
 
 import DirectBinaryUploadOptions from './direct-binary-upload-options';
-import { DefaultValues } from './constants';
+import { DefaultValues, RegularExpressions } from './constants';
+import UploadError from './upload-error';
+import ErrorCodes from './error-codes';
 
 /**
  * Options specific to a file system upload. Also supports all options defined by
  * DirectBinaryUploadOptions.
  */
 export default class FileSystemUploadOptions extends DirectBinaryUploadOptions {
 
+    constructor() {
+        super();
+        this.replaceValue = '-';
+        this.folderNodeProcessor = async (name) => {
+            return name.replace(RegularExpressions.INVALID_FOLDER_CHARACTERS_REGEX,
```

Comment:
Note: Could we end up with two different folders getting the same name because of characters replacement?

Reply:
Theoretically, yes. I think it's just going to be a side-effect of the requirement to do this name clean-up. I'll pass it up the chain. It won't cause any technical issues, but it's definitely worth communicating.